dali/operators/imgcodec/image_decoder.h

```diff
@@ -668,6 +668,10 @@ class ImageDecoder : public StatelessOperator<Backend> {
     tp_ = GetThreadPool(ws);
     assert(tp_ != nullptr);
     auto auto_cleanup = AtScopeExit([&] {
+      // Wait for all threads to finish. Even if we wait for the future, we are not guaranteed that
+      // the thread pool is still not in use, so we forcefully wait for all threads to finish before
+      // clearing the thread pool pointer.
+      tp_->WaitForWork();
```

Comment:
for safety, we want to make sure that tp_ has finished completely before clearing the pointer (we don't always control the lifetime of the thread pool as it can come from the workspace)